web/client-api/src/main/java/io/deephaven/web/client/api/JsTable.java

```diff
@@ -805,16 +810,48 @@ private JsTotalsTableConfig getTotalsDirectiveFromOptionalConfig(Object config)
                 }
                 workerConnection.tableServiceClient().asOfJoinTables(request, metadata, c::apply);
             };
-
-        } else if (Js.asPropertyMap(JoinTablesRequest.Type).has(joinType)){
+        } else if (joinType.equals("CROSS_JOIN")) {
```

Comment:
well now that we're unrolling this rather than sending the join type to the server, its probably time to consider breaking this method up into several, or publishing as part of JS the various join types, rather than just letting the server figure it out, and allowing the server to add new join types without client changes